src/socket.c

```diff
@@ -631,8 +631,13 @@ static void handle_input_filename(const gchar *buf)
 	{
 		if (g_str_has_suffix(locale_filename, ".geany"))
 		{
-			if (project_ask_close())
-				main_load_project_from_command_line(locale_filename, TRUE);
+			/* Don't open project when it is already opened */
+			if (!app->project || !(g_strcmp0(app->project->file_name, locale_filename) == 0
```

Comment:
Do you need to compare to the locale filename?  IIUC the `project->file_name` has to be UTF-8 or GTK display will be broken, so just compare to UTF-8 filename.